### PR TITLE
update to hashing method

### PR DESCRIPTION
**Pull request by kianwei96**
uses new hashlib functions by default, falls back to old version if can't find file to load

### CODE REVIEW

File: DataProcessingTools/objects.py [hunk 1]
```diff
@@ -73,18 +73,28 @@ def __init__(self, *args, **kwargs):
...
-                if self.dirs and saveLevel > 0:
-                    self.save()
+                # attempt old hashing - os/version dependent
+                fname = self.get_filename(legacy=1)
+                print('file name expected (old version): ', fname)
+                if redoLevel == 0 and os.path.isfile(fname):
+                    self.load(fname)
                     if verbose > 0:
-                        print("Object saved to file {0}".format(fname))
+                        print("Object loaded from file {0}".format(fname))
```
Comment (kianwei96) [inline DataProcessingTools/objects.py:89]: by default, we now get the hash of the arguments using hashlib purely (without mixing in hickle). if it is not able to find, before proceeding to create a new object it tries to get another hash value, using the old hickle method.

File: DataProcessingTools/objects.py [hunk 2]
```diff
@@ -200,19 +210,23 @@ def append(self, obj):
             for d in obj.dirs:
                 self.dirs.append(d)
 
-    def get_filename(self):
+    def get_filename(self,legacy=0):
```
Comment (kianwei96) [inline DataProcessingTools/objects.py:213]: new argument in get_filename to toggle between hashmodes, defaults to new

File: DataProcessingTools/objects.py (hunk 1 above)
Comment (kianwei96) [reply DataProcessingTools/objects.py:89]: object will always be saved with new hash

**Comment by grero** [comment]
Looks good. We just need to update the test to use the new filename when testing the hash.

**Comment by grero** [comment]
Test failures are all due to changing in hashing. I'll merge this first, then add another PR fixing the tests.

**Pull request merged**